src/segment_kubectl.go

```diff
@@ -31,15 +51,80 @@ func (k *kubectl) init(props *properties, env environmentInfo) {
 }
 
 func (k *kubectl) enabled() bool {
+	parseKubeConfig := k.props.getBool(ParseKubeConfig, false)
+	if parseKubeConfig {
+		return k.doParseKubeConfig()
+	} else {
+		return k.doCallKubectl()
+	}
+}
+
+func (k *kubectl) doParseKubeConfig() bool {
+	// Follow kubectl search rules (see https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable)
+	// TL;DR: KUBECONFIG can contain a list of files. If it's empty ~/.kube/config is used. First file in list wins when merging keys.
+	kubeconfigs := filepath.SplitList(k.env.getenv("KUBECONFIG"))
+	if len(kubeconfigs) == 0 {
+		kubeconfigs = []string{filepath.Join(k.env.homeDir(), ".kube/config")}
+	}
+	contexts := make(map[string]KubeConfigContext)
+	k.Context = ""
+	for _, kubeconfig := range kubeconfigs {
+		if kubeconfig == "" {
+			continue
+		}
+
+		content := k.env.getFileContent(kubeconfig)
+
+		var config KubeConfig
+		err := yaml.Unmarshal([]byte(content), &config)
+		if err != nil {
+			continue
+		}
+
+		for _, context := range config.Contexts {
+			if _, exists := contexts[context.Name]; !exists {
+				contexts[context.Name] = context.Context
+			}
+		}
+
+		if k.Context == "" {
+			k.Context = config.CurrentContext
+		}
+
+		context, exists := contexts[k.Context]
+		if exists {
+			k.KubeConfigContext = context
+			return true
+		}
+	}
+
+	displayError := k.props.getBool(DisplayError, false)
+	if displayError {
+		message := "KUBECONFIG ERR"
+		if k.Context == "" {
+			k.Context = message
+		}
+		k.Namespace = message
+		k.User = message
+		k.Cluster = message
+		return true
+	} else {
+		return false
+	}
+}
+
+func (k *kubectl) doCallKubectl() bool {
 	cmd := "kubectl"
 	if !k.env.hasCommand(cmd) {
 		return false
 	}
-	result, err := k.env.runCommand(cmd, "config", "view", "--minify", "--output", "jsonpath={..current-context},{..namespace}")
+	result, err := k.env.runCommand(cmd, "config", "view", "--minify", "--output", "jsonpath={..current-context},{..namespace},{..context.user},{..context.cluster}")
 	displayError := k.props.getBool(DisplayError, false)
 	if err != nil && displayError {
 		k.Context = "KUBECTL ERR"
 		k.Namespace = k.Context
+		k.User = k.Context
```

Comment:
This block and the above statement are identical, it might be useful to have a simple helper function to set the error context?

Reply:
I extracted it into a helper as suggested.